### PR TITLE
added pure, pureT, pureTT and pureTTT for all Applicatives

### PR DESCRIPTION
**Pull request by ChrisMyrants**

### CODE REVIEW

File: Sources/FunctionalKit/TypeConstructor.swift [hunk 1]
```diff
@@ -2,6 +2,28 @@ public protocol TypeConstructor {
 	associatedtype ParameterType
 }
 
+public protocol PureConstructible: TypeConstructor {
+    static func pure(_ value: ParameterType) -> Self
+}
+
+extension PureConstructible where ParameterType: PureConstructible {
+    static func pureT(_ value: ParameterType.ParameterType) -> Self {
```
Comment (broomburgo) [inline Sources/FunctionalKit/TypeConstructor.swift]: needs `public`

File: Sources/FunctionalKit/TypeConstructor.swift [hunk 2]
```diff
@@ -2,6 +2,28 @@ public protocol TypeConstructor {
...
+        return Self.pure(ParameterType.pure(value))
+    }
+}
+
+extension PureConstructible where ParameterType: PureConstructible, ParameterType.ParameterType: PureConstructible {
+    static func pureTT(_ value: ParameterType.ParameterType.ParameterType) -> Self {
+        return Self.pureT(ParameterType.ParameterType.pure(value))
```
Comment (broomburgo) [inline Sources/FunctionalKit/TypeConstructor.swift]: please just use `.pure`

**Review by broomburgo — APPROVED** [review]

File: Sources/FunctionalKit/TypeConstructor.swift (hunk 2 above)
Comment (ChrisMyrants) [reply Sources/FunctionalKit/TypeConstructor.swift]: done

File: Sources/FunctionalKit/TypeConstructor.swift (hunk 1 above)
Comment (ChrisMyrants) [reply Sources/FunctionalKit/TypeConstructor.swift]: done

**Review by RPallas92 — APPROVED** [review]

**Pull request merged**